src/main/java/com/Etech/Configuration/AppConfig.java

```diff
@@ -33,7 +31,7 @@ public ModelMapper modelMapper() {
         Converter<Map<Product, Integer>, List<ProductQuantityDto>> productMapConverter = ctx -> {
             Map<Product, Integer> source = ctx.getSource();
             return source.entrySet().stream()
-                    .map(entry -> new ProductQuantityDto(mapper.map(entry.getKey(), ProductDto.class), entry.getValue()))
+                    .map(entry -> new ProductQuantityDto(mapper.map(entry.getKey(), ProductWithOutQuantityDetailDto.class), entry.getValue()))
```

Comment:
use ProductWithOutQuantityDetailDto instead of Product DTO in order to hide details like product quantity while returning ProductDto